### PR TITLE
QSP-1220 Script to create github releases

### PR DESCRIPTION
**Pull request by arshdeeptinna**
Create github release whenever version.md is updated 
## Description

Release will be triggered only for develop branch. Currently the body of release will be a simple list of commits since last version which can be changed to have a summary section and list all commits under changes section. Script will also attach a bundle. 
Release create will be marked as draft and pre-release so that someone can review the bundle make any changes and then publish it. 

The example draft release created using version 2.0.3-beta1 is https://github.com/quantstamp/qsp-protocol-node/releases/tag/untagged-d971f49e7ae9e1fd21ce
## Motivation and Context

To avoid creating bundle manually

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] I have updated the version in src/qsp_protocol_node/config/config.py

### CODE REVIEW

File: bin/create-github-release [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    echo "Release created successfully"
+    
+    UPLOAD_URL=`echo $RESPONSE | jq .'upload_url' -r | cut -f1 -d"{"`
+    echo $UPLOAD_URL
+    echo "Trying to upload release bundle"
+    curl --data-binary "@$ZIP_DIR/$ZIP_FILENAME" -H "Authorization: token $GITHUB_TOKEN" -H "Content-Type: application/zip" -X POST "$UPLOAD_URL?name=$ZIP_FILENAME"
+    echo "Upload successful"
+  fi
+fi
+echo "Not on develop branch. Skipping release.."
```
Comment (lpassos) [inline bin/create-github-release]: New line at the end?

File: bin/create-github-release [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    git log $"$LATEST_TAG"..HEAD --pretty=format:"%h %s" > $SCRIPT_DIR/CHANGELOG.md
+    body="$(cat $SCRIPT_DIR/CHANGELOG.md)"
+
+    echo "Creating new bundle"
+    #"$SCRIPT_DIR/create-bundle"
+    if [ ! -f "$ZIP_DIR/$ZIP_FILENAME" ]; then 
+      echo "Failed to create a bundle file: $ZIP_FILENAME"
+      exit 1
+    fi
+    echo "Bundle Created"
```
Comment (lpassos) [inline bin/create-github-release]: "Created" => "created"

File: bin/create-github-release [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+LATEST_TAG=$(git describe --tags --abbrev=0)
+readonly DIST_NAME="qsp-protocol-$CURRENT_VERSION"
+readonly ZIP_FILENAME="${DIST_NAME}.zip"
+readonly ZIP_DIR="$SCRIPT_DIR/../deployment/local/dist"
+
+if [ -f "$SCRIPT_DIR/response.json" ]; then
+  rm $SCRIPT_DIR/response.json
+fi
+
+echo $LATEST_TAG
```
Comment (lpassos) [inline bin/create-github-release]: Rule of thumb: put all variables under double quotes....

File: bin/create-github-release [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+echo "Travis branch: $TRAVIS_BRANCH"
+DEVELOP_BRANCH='quantstamp\/develop'
+CURRENT_VERSION=`PYTHONPATH=:"$SCRIPT_DIR/../src/qsp_protocol_node" python3 -c "from config import Config; c = Config(); print(c.node_version)"`
+LATEST_TAG=$(git describe --tags --abbrev=0)
+readonly DIST_NAME="qsp-protocol-$CURRENT_VERSION"
+readonly ZIP_FILENAME="${DIST_NAME}.zip"
+readonly ZIP_DIR="$SCRIPT_DIR/../deployment/local/dist"
+
+if [ -f "$SCRIPT_DIR/response.json" ]; then
+  rm $SCRIPT_DIR/response.json
```
Comment (lpassos) [inline bin/create-github-release]: "$SCRIPT_DIR/response.json"

File: bin/create-github-release [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+fi
+
+echo $LATEST_TAG
+if [[ ! "$TRAVIS_BRANCH" =~ "$DEVELOP_BRANCH" ]]; then
+  if [[ "$CURRENT_VERSION" != "$LATEST_TAG" ]]; then
+    echo "New version tag: $CURRENT_VERSION"
+    echo "Need to create a new release"
+    echo "CHANEGLOG.md file created"
+    git log $"$LATEST_TAG"..HEAD --pretty=format:"%h %s" > $SCRIPT_DIR/CHANGELOG.md
+    body="$(cat $SCRIPT_DIR/CHANGELOG.md)"
```
Comment (lpassos) [inline bin/create-github-release]: Again, put pathnames under quotes when using variables.

File: bin/create-github-release [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+      }' > "$SCRIPT_DIR"/github-body.md
+    echo "Github body written to $SCRIPT_DIR/github-body.md"
+
+    echo "Trying to create the release"
+    curl --data "@$SCRIPT_DIR"/github-body.md -H "Authorization: token $GITHUB_TOKEN" -X POST "https://api.github.com/repos/quantstamp/qsp-protocol-node/releases" >> $SCRIPT_DIR/response.json
+    RESPONSE=`cat $SCRIPT_DIR/response.json`
+    echo "Release created successfully"
+    
+    UPLOAD_URL=`echo $RESPONSE | jq .'upload_url' -r | cut -f1 -d"{"`
+    echo $UPLOAD_URL
```
Comment (lpassos) [inline bin/create-github-release]: Put in in double quotes

File: bin/create-github-release (hunk 1 above)
Comment (arshdeeptinna) [reply bin/create-github-release]: fixed!

File: bin/create-github-release (hunk 2 above)
Comment (arshdeeptinna) [reply bin/create-github-release]: fixed

File: bin/create-github-release (hunk 3 above)
Comment (arshdeeptinna) [reply bin/create-github-release]: fixed

File: bin/create-github-release (hunk 4 above)
Comment (arshdeeptinna) [reply bin/create-github-release]: fixed

File: bin/create-github-release (hunk 5 above)
Comment (arshdeeptinna) [reply bin/create-github-release]: fixed

File: bin/create-github-release (hunk 6 above)
Comment (arshdeeptinna) [reply bin/create-github-release]: fixed

File: .travis.yml [hunk 7]
```diff
@@ -17,6 +17,7 @@ after_script:
   - sed -i -e "s@/app/@$PWD/@g" ./tests/coverage/.coverage
   - cat ./tests/coverage/.coverage
   - coveralls
+  - ./bin/create-github-release
```
Comment (lpassos) [inline .travis.yml]: Please make sure this file is chmod-ed with '+x'

File: bin/create-github-release [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+    echo "Release created successfully"
+    
+    UPLOAD_URL=`echo $RESPONSE | jq .'upload_url' -r | cut -f1 -d"{"`
+    echo "$UPLOAD_URL"
+    echo "Trying to upload release bundle"
+    curl --data-binary "@$ZIP_DIR/$ZIP_FILENAME" -H "Authorization: token $GITHUB_TOKEN" -H "Content-Type: application/zip" -X POST "$UPLOAD_URL?name=$ZIP_FILENAME"
+    echo "Upload successful"
+  fi
+fi
+echo "Not on develop branch. Skipping release.."
```
Comment (lpassos) [inline bin/create-github-release]: Still missing a new line here.

**Comment by lpassos** [comment]
@arshdeeptinna Small changes. Please fix them prior to merging. Otherwise, looks good.

File: .travis.yml (hunk 7 above)
Comment (arshdeeptinna) [reply .travis.yml]: done

File: bin/create-github-release (hunk 8 above)
Comment (arshdeeptinna) [reply bin/create-github-release]: fixed

**Pull request merged**